QtCollider/QcApplication.cpp

```diff
@@ -104,6 +104,29 @@ QcApplication::~QcApplication() {
     _mutex.unlock();
 }
 
+void QcApplication::createMenu() {
+    _mainMenu = QSharedPointer<QMenuBar>(new QMenuBar(0));
+
+#ifdef Q_OS_MAC
+    // macOS registers cmd+q on menu bars by default. Here we register a handler for cmd+q that does nothing
+    // and we disable the Quit menu by default
+    QMenu* menu;
+    QAction* action;
+    action = new QAction(tr("Catch CMD+Q on MacOS X"));
+    action->setMenuRole(QAction::QuitRole);
+    QObject::connect(action, SIGNAL(triggered()), this, SLOT(onQuit()));
+    menu = new QMenu(tr("&File"));
+    menu->addAction(action);
+    _mainMenu->addMenu(menu);
+    action->setEnabled(false);
+#endif
+}
+
+void QcApplication::onQuit() {
+    printf("[QcApplication::onQuit] CMD+Q was caught by the interpreter. This is weird, it should not happen. \
+    Please file an issue at https://github.com/supercollider/supercollider/issues");
```

Comment:
this looks a bit odd, why have 5 spaces in a row? Probably what you mean is this:

```cpp
    printf("[QcApplication::onQuit] CMD+Q was caught by the interpreter. This is weird, it should not happen. "
    "Please file an issue at https://github.com/supercollider/supercollider/issues");
```

successive string literals (like `"abc" "def"`) are automatically concatenated by the compiler.

Reply:
also, i believe you want `qWarning()` here, IIUC printf will not even work here. see for instance https://github.com/supercollider/supercollider/blob/develop/QtCollider/QtDownload.cpp#L77

Reply:
Excellent ! The multiline string trick is good to know !